consensus/consensus_service.go

```diff
@@ -470,22 +470,21 @@ func (consensus *Consensus) getLeaderPubKeyFromCoinbase(header *block.Header) (*
 func (consensus *Consensus) UpdateConsensusInformation() Mode {
 	pubKeys := []*bls.PublicKey{}
 	hasError := false
-
 	header := consensus.ChainReader.CurrentHeader()
-
 	epoch := header.Epoch()
-	curPubKeys := core.CalculatePublicKeys(epoch, header.ShardID())
+	// currentSuperCommittee, _ := consensus.ChainReader.ReadShardState(epoch)
+	curPubKeys := consensus.CommitteeAssigner.ReadPublicKeys(epoch)
+	// curPubKeys := core.CalculatePublicKeys(epoch, header.ShardID(), committee.GenesisAssigner, currentSuperCommittee)
 	consensus.numPrevPubKeys = len(curPubKeys)
-
 	consensus.getLogger().Info().Msg("[UpdateConsensusInformation] Updating.....")
 
 	if core.IsEpochLastBlockByHeader(header) {
 		// increase epoch by one if it's the last block
 		consensus.SetEpochNum(epoch.Uint64() + 1)
 		consensus.getLogger().Info().Uint64("headerNum", header.Number().Uint64()).
 			Msg("[UpdateConsensusInformation] Epoch updated for next epoch")
-		nextEpoch := new(big.Int).Add(epoch, common.Big1)
-		pubKeys = core.CalculatePublicKeys(nextEpoch, header.ShardID())
+		// nextEpoch := new(big.Int).Add(epoch, common.Big1)
```

Comment:
Why change the logic to not read for next epoch?

Reply:
Because I didn't get around to asking question: What exactly is meaning of giving me committee of next epoch in staking? I don't know the next epoch's commitee until beacon chain gives it to me, and non shard-0 nodes presumably call this.